coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -429,6 +429,33 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
     PlanResult(execPlan::Nil)
   }
 
+  private def materializeTsCardinalities(qContext: QueryContext, lp: TsCardinalities) : PlanResult = {
+    // This code is nearly identical to materializeMetadataQueryPlan, but it prevents some
+    //   boilerplate code clutter that results when TsCardinalities extends MetadataQueryPlan.
+    //   If this code's maintenance isn't worth some extra stand-in lines of code (i.e. at matchers that
+    //   take MetadataQueryPlan instances), then just extend TsCardinalities from MetadataQueryPlan.
+    val queryParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+    val partitions = partitionLocationProvider.getAuthorizedPartitions(
+      TimeRange(queryParams.startSecs * 1000, queryParams.endSecs * 1000))
+    val execPlan = if (partitions.isEmpty) {
+      logger.warn(s"No partitions found for ${queryParams.startSecs}, ${queryParams.endSecs}")
+      localPartitionPlanner.materialize(lp, qContext)
+    } else {
+      val leafPlans = partitions.map { p =>
+        logger.debug(s"partitionInfo=$p; queryParams=$queryParams")
+        if (p.partitionName.equals(localPartitionName))
+          localPartitionPlanner.materialize(lp, qContext)
+        else {
+          createMetadataRemoteExec(qContext, p, Map("match[]" -> queryParams.promQl))
```

Comment:
This is following suit with `LabelNames` and `LabelCardinality`. I'm not 100% sure this is right, though, and I need to figure out how to test whether-or-not it works

Reply:
deferred to later PR